fdbserver/StorageCache.actor.cpp

```diff
@@ -1969,15 +1970,73 @@ ACTOR Future<Void> storageCacheServer(StorageServerInterface ssi, uint16_t id, R
 	state StorageCacheData self(ssi.id(), id, db);
 	state ActorCollection actors(false);
 	state Future<Void> dbInfoChange = Void();
+	state StorageCacheUpdater updater(self.lastVersionWithData);
+	self.updater = &updater;
+	state Transaction tr(self.cx);
+	state Value trueValue = storageCacheValue(std::vector<uint16_t>{ 0 });
+	state Value falseValue = storageCacheValue(std::vector<uint16_t>{});
+	state MutationRef privatized;
+	state Version readVersion;
 
 	//TraceEvent("StorageCache_CacheServerInterface", self.thisServerID).detail("UID", ssi.uniqueID).detail("IsCacheServer", ssi.isCacheServer);
+
 	// This helps identify the private mutations meant for this cache server
 	self.ck = cacheKeysPrefixFor( id ).withPrefix(systemKeys.begin);  // FFFF/02cacheKeys/[this server]/
 
 	actors.add(waitFailureServer(ssi.waitFailure.getFuture()));
 	actors.add(traceCounters("CacheMetrics", self.thisServerID, SERVER_KNOBS->STORAGE_LOGGING_DELAY, &self.counters.cc, self.thisServerID.toString() + "/CacheMetrics"));
 
-	// compactCache actor will periodically compact the cache when certain version condition is met
+	TraceEvent(SevDebug, "SCStartUpPostWarmup", self.thisServerID).detail("DataVersionBefore", self.version.get());
```

Comment:
I think this whole block that follows should be in its own actor - you can still wait on it, but having another actor would make this way more readable

Reply:
That's exactly what I started with. Having the new code block in a new actor. But, don't we want this code to execute serially (esp the fetching of metadata?) before we schedule other actors? That's why I ended up moving it inline.

Reply:
Right, but if you call wait on your actor before you continue, nothing else that follows that wait-call will run concurrently

Reply:
Ah ok, makes sense. Duh!